crates/astria-sequencer-relayer/src/main.rs

```diff
@@ -44,10 +51,30 @@ async fn main() -> ExitCode {
         "initializing sequencer relayer"
     );
 
-    SequencerRelayer::new(cfg)
-        .expect("could not initialize sequencer relayer")
-        .run()
-        .await;
+    let mut sigterm = signal(SignalKind::terminate())
+        .expect("setting a SIGTERM listener should always work on Unix");
+    let (sequencer_relayer, shutdown_handle) =
+        SequencerRelayer::new(cfg).expect("could not initialize sequencer relayer");
+    let sigterm_handle = tokio::spawn(async move {
+        let shutdown_token = shutdown_handle.token();
+        tokio::select!(
+            _ = sigterm.recv() => {
+                // We don't care about the result (i.e. whether there could be more SIGTERM signals
+                // incoming); we just want to shut down as soon as we receive the first `SIGTERM`.
+                info!("received SIGTERM, issuing shutdown to all services");
+                shutdown_handle.shutdown();
+            }
+            () = shutdown_token.cancelled() => {
+                // Nothing to log here - we're just finishing up waiting for a SIGTERM having
```

Comment:
Can you report this at a `warn!` level? This arm should never match - if it does match, then something strange is going on inside sequencer relayer (i.e. it shouldn't just shut itself down).